app/helpers/application_helper.rb

```diff
@@ -15,7 +15,10 @@ def l(date, options = {})
   def decorate(object, decorator_class = nil)
     decorator_class ||= "#{object.class.to_s.demodulize}Decorator".constantize
     decorator = decorator_class.new(object, self)
-    yield(decorator) if block_given?
+
+    yield(decorator) if block_given? && decorator.present?
+    return nil if decorator.blank?
```

Comment:
Possibly my misunderstanding.

Isn't the purpose of the Null Object Pattern so that you do not need to do these checks (`decorator.present?` and `decorator.blank?`), and the null instance can be used as normal?

Reply:
Only the helper relies on the NilClassDecorator to explicitly return a `nil`. In the views a decorated object which was `nil` will therefore remain `nil`

Sadly a nil decorator object is not `nil`   - it is an instance of `NilClassDecorator(nil)` which is always truthy - only the `nil` object itself is "falsey"

```
NilClassDecorator(nil) => truthy
BUT
NilClassDecorator(nil).nil? => true
NilClassDecorator(nil).blank? => true
NilClassDecorator(nil).present? => false
```

Reply:
quite a useful [summary of ruby nil here](https://www.rubyguides.com/2018/01/ruby-nil/)